app/src/main/java/nu/yona/app/api/manager/dao/ActivityTrackerDAO.java

```diff
@@ -92,12 +85,26 @@ public void saveActivities(Activity activity)
 		return activityList;
 	}
 
+
 	/**
 	 * Clear activities.
 	 */
 	public void clearActivities()
 	{
-		delete(DBConstant.TBL_ACTIVITY_TRACKER, null, null);
-		getActivities();
+		delete(DBConstant.TBL_ACTIVITY_TRACKER,
+				DBConstant.APPLICATION_START_TIME +
+						" IN (SELECT " + DBConstant.APPLICATION_START_TIME +
+						" from " + DBConstant.TBL_ACTIVITY_TRACKER +
+						" ORDER BY " + DBConstant.APPLICATION_NAME + " DESC LIMIT " + DBConstant.ACTIVITY_FETCH_ROW_LIMIT + ")",
+				null);
+		printTotalActivityCount();
 	}
+
+	public void printTotalActivityCount()
```

Comment:
This doesn't print but log, so it should be named ``logTotalActivityCount``. I missed that in the earlier round. I'll push an update with the rename.